File: src/i1tbfssymmetric/Go/SymmetricTree.go
package main

func isSymmetric(root *TreeNode[int]) bool {
	queue := &Queue{}
	queue.Enqueue(root.Left)
	queue.Enqueue(root.Right)

	for queue.Size() > 0 {
		left := queue.Dequeue().(*TreeNode[int])
		right := queue.Dequeue().(*TreeNode[int])

		if left == nil && right == nil {
			continue
		}
		if left == nil || right == nil {
			return false
		}
		if left.Data != right.Data {
			return false
		}

		queue.Enqueue(left.Left)
		queue.Enqueue(right.Right)
		queue.Enqueue(left.Right)
		queue.Enqueue(right.Left)
	}
	return true
}

File: src/i1tbfssymmetric/Python/SymmetricTree.py
from Queue import Queue
from TreeNode import TreeNode


def is_symmetric(root):
    queue = Queue()
    queue.enqueue(root.left)
    queue.enqueue(root.right)

    while queue.size > 0:
        left = queue.dequeue()
        right = queue.dequeue()

        if left is None and right is None:
            continue
        if left is None or right is None:
            return False
        if left.data != right.data:
            return False

        queue.enqueue(left.left)
        queue.enqueue(right.right)
        queue.enqueue(left.right)
        queue.enqueue(right.left)

    return True
